### DISCUSSION TITLE
How to specify the dependent tests

### DISCUSSION THREAD

**Discussion by senketsu03** [nunit/nunit#4411] [Q&A] [answered]
No one comes into [discussions page](https://github.com/nunit/nunit/discussions/4397), so I decided to ask here. Duplicating my question:

`[Order]` attribute isn't very helpful in my case. I'm writing integration tests for my db. Suppose I have a table `users`, `courses` and `users_courses`. Every table has it's own class for testing, e.g. `UsersTests.cs`, `CoursesTests.cs` and `UsersCourses.cs`.
As the name suggests, the `users_courses` table depends on PKs from `users` and `courses` tables, therefore the test to push data into `users_courses` table before tests from `UsersTests` and `CoursesTests` pushes data into correspodning tables fails. `[Order]` attribute applies only for methods in one class, so there's no point in using it in my case at all.

I tried implementing the `TestState` and `TestKind` classes, checking is needed test was run in the beginning of the test code, e.g.:

```c#
while (TestState.CheckCompelted(TestStates, TestKind.PushUsers) == false)
{
    await Task.Delay(1);
}
```

And added the `[Parallelizable]` attribute to a class, but since I'm using in-memory sqlite database it doesn't support parallel transactions, so everything fails.

If there's an alternative solution for my problem, please, let me know.

**Comment by OsirisTerje** [comment, marked as answer]
I'm sorry, but this is still not an issue, it belongs in the discussion.  If you want to ping people, do that instead.  Our names are all over the place, so take a wild guess and ask directly - on the discussion page, OR, go to our slack channel and ask there. 

And, we do read the discussions and try to answer, but....

The reason for you not getting a response is that first, your question is unclear (what are you actually trying to test here?), and second doesn't really have anything to do with NUnit (it is more about how you design a good test). 

And, if you add a repro solution, that could make it more clear what you're trying to achieve.  

If I'm trying to read what you write above (which doesn't really make sense to me, but that might be just me):
How do you access your db, why do you have separate tables for testing - are you not testing the system itself?  
Designing a db test setup that require you to do something that is not done in production kind of defeats the purpose. So, why not just populate the db with a bare minimum.  Do the operations that your system should normally do, and verify that it works as intended.  A test should be atomic and isolated.  
Since you're using in-memory database, you really don't need to worry about sequence.  Just write to the first db table, check the result, kill the db. Next test, same with the 2nd table, kill it after test, then continue with in between one, but now you can do that with the 2 first tables filled up. And so on...  No need for sequences.

**Comment by mikkelbu** [comment]
@senketsu03 Also please note that all of us do this in our spare time (for free), so if nobody answers immediately then it can also just be that people are busy (and personally I get around 50 notifications a day for different open source projects, so sometimes it is a fulltime job just to keep up)
